proposals/public-suffix.md

```diff
@@ -0,0 +1,836 @@
+from a UX perspective to group them by their registrable domains.
+
+##### Example
+
+| No grouping           | ---------> | With grouping     |                       |
+|-----------------------|------------|-------------------|-----------------------|
+| example.co.uk         |            | **example.co.uk** |                       |
+| example2.com          |            |                   | example.co.uk         |
+| foo.bar.example.co.uk |            |                   | foo.bar.example.co.uk |
+| foo.bar.example2.com  |            |                   | www.example.co.uk     |
+| www.example.co.uk     |            |                   |                       |
+| www.example2.com      |            | **example2.com**  |                       |
+|                       |            |                   | example2.com          |
+|                       |            |                   | foo.bar.example2.com  |
+|                       |            |                   | www.example2.com      |
+
+#### 3. Detect Third-Party Requests
+
+Some ad-blocker extensions allow users to create request-blocking rules that only
+apply to *[third-party requests](https://help.adblockplus.org/hc/en-us/articles/360062733293-How-to-write-filters#party-requests)*. Additionally, tracker-blocking extensions
+typically exclude requests from tracking detection unless they are third-party requests.
+
+In order to determine if a request is a third-party request, such extensions lookup
+the registrable domain of the request and of the parent document using the PSL.
+If the registrable domain of the request is different to that of the parent
+document, then the request is considered third-party.
+
+##### Example of Determining if Requests are Third-Party
+
+| Req. | Domain                      | Registrable Domain          | Third-party? |
+|:----:|----------------------------:|----------------------------:|:------------:|
+| #1   | foo.com                     | foo.com                     |              |
+| #2   | bar.com                     | bar.com                     | Yes          |
+|      |                             |                             |              |
+| #1   | foo.amazonaws.com           | amazonaws.com               |              |
+| #2   | bar.amazonaws.com           | amazonaws.com               | No           |
+|      |                             |                             |              |
+| #1   | foo.amazonaws.com           | amazonaws.com               |              |
+| #2   | bar.us-east-1.amazonaws.com | bar.us-east-1.amazonaws.com | Yes          |
+
+#### 4. Search vs Navigate
+
+Firefox makes use of the PSL in order to determine whether to issue a search query
+or whether to try a navigation, when a user enters a domain-like string in the
+url bar. In such instances, a PSL lookup is made and:
+
+* If the domain has a known eTLD, attempt to navigate.
+* If the domain has an unknown eTLD, use a search engine.
```

Comment:
This is a use case for the PSL, but I'm not sure it's as compelling a use case for extensions themselves (since it boils down to "handle things with TLDs differently", which is already described)

Reply:
This use case was provided by a reviewer during review of this proposal to justify the inclusion of functionality to distinguish between known vs unknown suffixes. I will add an update to note that this specific use case may not apply directly to extensions.

Reply:
@rdcronin IMHO there are two ways the PSL is used:

1. Figuring out a site, for grouping or associating data with a site, or deleting site history, etc.
2. Figuring out if something looks like a valid domain - that's what this is and I don't think it's covered above...? Actually, it looks like all the other examples are examples of (1.) and this is the only example of (2.).